### PR TITLE
Minor `TypeSystem/Schema` cleanup.

### PR DESCRIPTION
**Pull request by mzhurovich**

### CODE REVIEW

File: TypeSystem/Schema/smoke_test_struct.h [hunk 1]
```diff
@@ -52,20 +46,7 @@ CURRENT_STRUCT(FullTest) {
...
-  CURRENT_FIELD(p, t_p);
-  CURRENT_FIELD(o, Optional<Primitives>);
-  typedef Variant<A, B, C, Empty> t_q;
-  CURRENT_FIELD(q, t_q);
-  CURRENT_CONSTRUCTOR(FullTest)(Variant<A, B, C, Empty> && q) : q(std::move(q)) {}
-};
-#endif
 
-}  // namespace
+}  // namespace SMOKE_TEST_STRUCT_NAMESPACE
```
Comment (dkorolev) [inline TypeSystem/Schema/smoke_test_struct.h]: `SMOKE_TEST_STRUCT_NAMESPACE` is a `#define`-d symbol, a bit misleading to but it here as a comment IMHO.

**Comment by dkorolev** [comment]
@mzhurovich if it passes the test now, I wonder why. And why remove the "should compile" section, it was here for a reason.

LGTM to merge.

File: TypeSystem/Schema/smoke_test_struct.h (hunk 1 above)
Comment (mzhurovich) [inline TypeSystem/Schema/smoke_test_struct.h]: Why? It explains what namespace is closed here.

Comment (dkorolev) [inline TypeSystem/Schema/smoke_test_struct.h]: Perhaps elaborate then, ex.
```
}  // namespace SMOKE_TEST_STRUCT_NAMESPACE, where `SMOKE_TEST_STRUCT_NAMESPACE` is `#define`-d above.
```

**Comment by mzhurovich** [comment]
@dkorolev Why it shouldn't? And was that particular  `Clone` test useful?

**Comment by dkorolev** [comment]
@mzhurovich I recall we had troubles with copy constructor disabled, and type elision not working for this type. Hence `Clone()`-s everywhere.

I'm worried this test did not pass on my machine. Let's sync up on this off the pull request.

File: TypeSystem/Schema/smoke_test_struct.h (hunk 1 above)
Comment (mzhurovich) [inline TypeSystem/Schema/smoke_test_struct.h]: Maybe rename the namespace to `SMOKE_TEST_STRUCT`?

Comment (dkorolev) [inline TypeSystem/Schema/smoke_test_struct.h]: It has to be a `#define`-d symbol to enable `#include`-ing this `smoke_test_struct.h` multiple times.

**Comment by dkorolev** [comment]
Merging https://github.com/C5T/Current/pull/441 and https://github.com/C5T/Current/pull/442.

**Pull request merged**